src/test/java/org/rulez/demokracia/pdengine/BeatTableCompareBeatsTest.java

```diff
@@ -0,0 +1,70 @@
+package org.rulez.demokracia.pdengine;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.rulez.demokracia.pdengine.annotations.TestedBehaviour;
+import org.rulez.demokracia.pdengine.annotations.TestedFeature;
+import org.rulez.demokracia.pdengine.annotations.TestedOperation;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.pdengine.testhelpers.CreatedBeatTable;
+
+@TestedFeature("Schulze method")
+@TestedOperation("compare beats")
+public class BeatTableCompareBeatsTest extends CreatedBeatTable {
+
+	@Before
+	public void setUp() throws ReportedException {
+		super.setUp();
+	}
+
+	@TestedBehaviour("if beats are different, the bigger wins")
+	@Test
+	public void compareBeats_throws_an_exception_when_first_input_param_is_not_defined() {
+		assertThrows(() -> beatTable.compareBeats(beats1, null)
+				).assertMessageIs("Invalid Pair key");
+	}
+	
+	@TestedBehaviour("if beats are different, the bigger wins")
+	@Test
+	public void compareBeats_throws_an_exception_when_second_input_param_is_not_defined() {
+		assertThrows(() -> beatTable.compareBeats(null, beats2)
+				).assertMessageIs("Invalid Pair key");
+	}
+	
+	@TestedBehaviour("if beats are different, the bigger wins")
+	@Test
+	public void compareBeats_gives_back_the_forward_bigger_beat1() {	
+		setCompareBeatsAsResult(beats1, beats2);
+		assertTrue(result.winning == beats1.winning && result.losing == beats1.losing);
+	}
+	
+	@TestedBehaviour("if beats tie, looses decide")
+	@Test
+	public void compareBeats_gives_back_the_backward_lower_beat1() {
+		setCompareBeatsAsResult(beats1, beats3);	
+		assertTrue(result.winning == beats1.winning && result.losing == beats1.losing);
+	}
+	
+    @TestedBehaviour("Unimplemented")
+	@Test
+	public void compareBeats_is_not_yet_implemented_when_the_result_is_tie() {
+		assertThrows(() -> beatTable.compareBeats(beats1, beats1)
```

Comment:
use assertUnimplemented

Reply:
as I see there is no such a method like assertUnimplemented

Reply:
Ok I think I created that method I hope you will like it. :)